watertap/unit_models/electrodialysis_1D.py

```diff
@@ -1556,6 +1691,165 @@ def eq_dl_thickness(self, mem, side, t, x):
                     * self.current_dens_lim_x[t, x] ** -1
                 )
 
+    def _get_fluid_dimensionless_quantities(self):
+        self.general_mass_diffusivity = Var(
+            initialize=1e-9,
+            bounds=(1e-16, 1e-6),
+            units=pyunits.meter**2 * pyunits.second**-1,
+            doc="The mass diffusivity of the solute as molecules (not individual ions)",
+        )
+        self.hydraulic_diameter = Var(initialize=1e-3, units=pyunits.meter)
+        self.N_Re = Var(
+            initialize=100, units=pyunits.dimensionless, doc="Reynolds Number"
+        )
+        self.N_Sc = Var(
+            initialize=2000, units=pyunits.dimensionless, doc="Schmidt Number"
+        )
+        self.N_Sh = Var(
+            initialize=100, units=pyunits.dimensionless, doc="Sherwood Number"
+        )
+
+        if self.config.hydraulic_diameter_method == HydraulicDiameterMethod.fixed:
+            _log.warning("Do not forget to FIX the channel hydraulic diameter in [m]!")
+        else:
+
+            @self.Constraint(
+                doc="To calculate hydraulic diameter",
+            )
+            def eq_hydraulic_diameter(self):
+                if (
+                    self.config.hydraulic_diameter_method
+                    == HydraulicDiameterMethod.conventional
+                ):
+                    return (
+                        self.hydraulic_diameter
+                        == 2
+                        * self.channel_height
+                        * self.cell_width
+                        * self.spacer_porosity
+                        * (self.channel_height + self.cell_width) ** -1
```

Comment:
I think we should remove this and only use the one below.

Reply:
I kept this as an option  for now because several papers used this equation. Maybe I can use this equation in some cases for comparison but I can also remove it in a later PR if ok.